File: Diff5/Problem 25 inc/Fibonacci.java
public class Fibonacci {
    public static void main(String[] commandLine) {
        boolean hasThree = false;
        int place = 1;
        for(int i=1; !hasThree;i++){
            hasThree = digitCount(toFib(i),1000);
            place++;
        }
        System.out.println(place);
    }
    public static int toFib(int index) {
        if (index == 1) return 1;
        if (index == 2) return 1;
        else return toFib(index - 2) + toFib(index - 1);
    }
    public static boolean digitCount(int num, int digits) {
        return num / Math.pow(10, digits - 1) >= 1;
    }
}

File: 05diff/25problem/Fibonacci.py
import time 
def toFib(index):
    sub2 = 0
    sub1 = 1
    for i in range(1,index):
        storage = sub1
        sub1 += sub2
        sub2 = storage 
    return sub1 
     
def solution():
    place = 1
    while not digitCount(toFib(place), 1000 ): 
        place += 1
    print place, toFib(place)
   
def digitCount(number, digits):
    return len(str(number)) == digits 
    
start = time.time()
solution()
print time.time() - start 
